dandi/delete.py

```diff
@@ -0,0 +1,220 @@
+            raise ValueError("Cannot delete assets from multiple Dandisets at once")
+
+    def register_dandiset(self, api_url: str, dandiset_id: str) -> None:
+        self.set_dandiset(api_url, dandiset_id)
+        self.deleting_dandiset = True
+
+    def register_asset(
+        self, api_url: str, dandiset_id: str, version_id: str, asset_path: str
+    ) -> None:
+        if version_id != "draft":
+            raise ValueError("Cannot delete assets from published versions")
+        self.set_dandiset(api_url, dandiset_id)
+        asset = self.client.get_asset_bypath(dandiset_id, version_id, asset_path)
+        if asset is None:
+            raise NotFoundError(
+                f"Asset at path {asset_path!r} not found in Dandiset {dandiset_id}"
+            )
+        self.remote_assets.add(
+            RemoteAsset(dandiset_id, version_id, asset["asset_id"], asset["path"])
+        )
+
+    def register_asset_folder(
+        self, api_url: str, dandiset_id: str, version_id: str, folder_path: str
+    ) -> None:
+        if version_id != "draft":
+            raise ValueError("Cannot delete assets from published versions")
+        self.set_dandiset(api_url, dandiset_id)
+        any_assets = False
+        for asset in self.client.get_dandiset_assets(
+            dandiset_id, version_id, path=folder_path
+        ):
+            self.remote_assets.add(
+                RemoteAsset(dandiset_id, version_id, asset["asset_id"], asset["path"])
+            )
+            any_assets = True
+        if not any_assets:
+            raise NotFoundError(
+                f"No assets under path {folder_path!r} found in Dandiset {dandiset_id}"
+            )
+
+    def register_url(self, url: str) -> None:
+        server_type, server_url, asset_type, asset_id = parse_dandi_url(url)
+        if asset_type == "dandiset":
+            if asset_id.get("version") is not None:
+                raise NotImplementedError(
+                    "Dandi API server does not support deletion of individual"
+                    " versions of a dandiset"
+                )
+            self.register_dandiset(server_url, asset_id["dandiset_id"])
+        elif asset_type == "item":
+            self.register_asset(
+                server_url,
+                asset_id["dandiset_id"],
+                asset_id.get("version") or "draft",
+                asset_id["location"],
+            )
+        elif asset_type == "folder":
+            self.register_asset_folder(
+                server_url,
+                asset_id["dandiset_id"],
+                asset_id.get("version") or "draft",
+                asset_id["location"],
+            )
+        else:
+            raise RuntimeError(f"Unexpected asset type for {url}: {asset_type}")
+
+    def register_local_path_equivalent(self, instance_name: str, filepath: str) -> None:
+        instance = known_instances[instance_name]
+        if instance.metadata_version == 0:
```

Comment:
```suggestion
        if instance.girder:
```
to make it more explicit than some possible relationship between having metadata version and girder